improve win32_event_log docs (#12755)
DataDog/integrations-core: win32_event_log/README.md

## Overview

The Win32 event log check watches for Windows Event Logs and forwards them to Datadog. Enable this check to:

- Track system and application events in Datadog.
- Correlate system and application events with the rest of your application.

> See the [Windows Event Logs documentation][13].

Reviewer: ```suggestion
For more information, see the [Windows Event Logging documentation][13].
```

### Configuration

Windows Event logs can be collected as one or both of the following methods.
- As [Datadog Events][16]
- As [Datadog Logs][17]

Both methods are configured in `win32_event_log.d/conf.yaml` in the `conf.d/` folder at the root of your [Agent's configuration directory][2]. See the [sample win32_event_log.d/conf.yaml][3] for all available configuration options.

> First, identify the Windows Event Log channels you want to monitor.

Reviewer: ```suggestion
```
Moving this bit down

#### List Windows Event channels

> To see a list of channels, run the following command in PowerShell:

Reviewer: ```suggestion
First, identify the Windows Event Log channels you want to monitor. To see a list of channels, run the following command in PowerShell:
```

<!-- xxx tabs xxx -->
<!-- xxx tab "Events" xxx -->

#### Event collection

To collect Windows Event Logs as Datadog Events, configure channels under the `instances:` section of your `win32_event_log.d/conf.yaml` configuration file.

The agent can be configured to collect Windows Event Logs as Datadog Events in two ways. Each method has its own configuration syntax for channels and for filters (see [Filtering Events](?tab=events#filtering-events)). The legacy method uses WMI and is the default mode for an instance. The newer method uses the Event Log API. We recommend using the Event Log API because it has better performance. To use the Event Log API collection method, set `legacy_mode: false` in each instance.

This example shows entries for the `Security` and `<CHANNEL_2>` channels:

```yaml
init_config:
instances:
  - # WMI - Legacy mode (default)
    legacy_mode: true
    log_file: Security

  - # Event Log API (better performance)
    path: Security
    legacy_mode: false
    filters: {}

  - path: "<CHANNEL_2>" 
    legacy_mode: false
    filters: {}
```

<!-- xxz tab xxx -->
<!-- xxx tab "Logs" xxx -->

#### Log collection

_Available for Agent versions >6.0_

> Log collection is disabled by default in the Datadog Agent; ensure to [activate log collection][18] (e.g. set `logs_enabled: true` in your `datadog.yaml` file) in order to collect Windows Event Logs as Datadog Logs.

Reviewer: ```suggestion
Log collection is disabled by default in the Datadog Agent. To collect Windows Event Logs as Datadog Logs, [activate log collection][18] by setting `logs_enabled: true` in your `datadog.yaml` file.
```

  - type: windows_event
    channel_path: "<CHANNEL_2>"
    source: windows.events
    service: myservice
```

Set the corresponding `source` parameter to `windows.events` to benefit from the [integration automatic processing pipeline][5].

<!-- xxz tab xxx -->
<!-- xxz tabs xxx -->

Edit the `<CHANNEL_2>` parameters with the Windows channel name you want to collect events from.

Finally, [restart the Agent][4].

**Note**: For the Security logs channel, add your Datadog Agent user to the `Event Log Readers` user group.

### Filtering events

> 1. Configure one or more filters for the event log. A filter allows you to choose what log events you want to get into Datadog.

Reviewer: ```suggestion
Configure one or more filters for the event log. A filter allows you to choose what log events you want to get into Datadog.
```

  <!-- xxx tabs xxx -->
  <!-- xxx tab "Events" xxx -->

  Use the Windows Event Viewer GUI to list all the event logs available for capture with this integration.

  To determine the exact values, set your filters to use the following PowerShell command:

  ```text
  Get-WmiObject -Class Win32_NTLogEvent
  ```

  For example, to see the latest event logged in the `Security` log file, use:

  ```text
  Get-WmiObject -Class Win32_NTLogEvent -Filter "LogFile='Security'" | select -First 1
  ```

  The values listed in the output of the command can be set in `win32_event_log.d/conf.yaml` to capture the same kind of events.

  <div class="alert alert-info">
  The information given by the  <code>Get-EventLog</code> PowerShell command or the Windows Event ViewerGUI may slightly differ from <code>Get-WmiObject</code>.<br>
  Double-check your filters' values with <code>Get-WmiObject</code> if the integration doesn't capture the events you set up.
  </div>

  Example legacy mode filters:

>   - log_file: Application, System, Setup, Security
>   - type: Critical, Error, Warning, Information, Audit Success, Audit Failure
>   - source_name: Any available source name
>   - event_id: Windows EventLog ID
> 
>   Example non-legacy mode filters:
> 
>   - path: Application, System, Setup, Security
>   - type: Critical, Error, Warning, Information, Success Audit, Failure Audit
>   - source: Any available source name
>   - id: event_id: Windows EventLog ID

Reviewer: ```suggestion
  - `log_file`: `Application`, `System`, `Setup`, `Security`
  - `type`: `Critical`, `Error`, `Warning`, `Information`, `Audit Success`, `Audit Failure`
  - `source_name`: Any available source name
  - `event_id`: Windows EventLog ID

  Example non-legacy mode filters:

  - `path`: `Application`, `System`, `Setup`, `Security`
  - `type`: `Critical`, `Error`, `Warning`, `Information`, `Success Audit`, `Failure Audit`
  - `source`: Any available source name
  - `id`: event_id: Windows EventLog ID
```

      - type: include_at_match
        name: application_errors_and_warnings
        pattern: '"level":"((?i)warning|error)"'
  ```

  Here is an example regex pattern to only collect Windows Events Logs from a certain EventID:

  ```yaml
  logs:
    - type: windows_event
      channel_path: Security
      source: windows.event
      service: Windows
      log_processing_rules:
        - type: include_at_match
          name: include_x01

>           pattern: \"value\":\"(101|201|301)\"

Reviewer 2: fixed pattern lost in rebase
```suggestion
          pattern: '"EventID":"(101|201|301)"'
```

  ```

  **Note**: The pattern may vary based on the format of the logs. The [Agent `stream-logs` subcommand][15] can be used to view this format.

  For more examples of filtering logs, see the [Advanced Log Collection documentation][12].

  #### Legacy events
  _Applies to Agent versions less than 7.41_

  Legacy Provider EventIDs have a `Qualifiers` attribute that changes the format of the log, as seen in the [Windows Event Schema][14]. These events have the following XML format, visible in Windows Event Viewer:
  ```xml
  <EventID Qualifiers="16384">3</EventID>
  ```

  The following regex must be used to match these EventIDs:
  ```yaml
  logs:
    - type: windows_event
      channel_path: Security
      source: windows.event
      service: Windows
      log_processing_rules:
        - type: include_at_match
          name: include_legacy_x01
          pattern: '"EventID":{"value":"(101|201|301)"'
  ```

  Agent versions 7.41 and later normalize the EventID field and this legacy pattern is no longer applicable.

<!-- xxz tab xxx -->
<!-- xxz tabs xxx -->

> 2. [Restart the Agent][4] using the Agent Manager (or restart the service).

Reviewer: ```suggestion
When you're done setting up filters, [restart the Agent][4] using the Agent Manager (or restart the service).
```
